api/src/test/java/org/openmrs/api/impl/DiagnosisServiceImplTest.java

```diff
@@ -330,5 +340,212 @@ public void purgeDiagnosis_shouldPurgeDiagnosis() {
+		assertEquals(diagnosis.getCreator(), savedDiagnosisAttributeType.getCreator());
+		assertEquals(diagnosis.getDateCreated(), savedDiagnosisAttributeType.getDateCreated());
+		assertEquals("353af72e-bb6e-4ed9-a1bf-0d8106ac2c15", savedDiagnosisAttributeType.getUuid());
+		assertEquals(false, savedDiagnosisAttributeType.getRetired());
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#saveDiagnosisAttributeType(DiagnosisAttributeType)
+	 */
+	@Test
+	public void saveDiagnosisAttributeType_shouldEditTheExistingDiagnosisAttributeType() {
+		DiagnosisAttributeType diagnosisAttributeType = diagnosisService.getDiagnosisAttributeTypeById(3);
+		assertEquals("Diagnostic Criteria", diagnosisAttributeType.getName());
+		diagnosisAttributeType.setName("Diagnosis Orientation");
+		diagnosisService.saveDiagnosisAttributeType(diagnosisAttributeType);
+		assertThat(diagnosisService.getDiagnosisAttributeTypeById(diagnosisAttributeType.getId()).getName(), is("Diagnosis Orientation"));
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getDiagnosisAttributeByUuid(String)
+	 */
+	@Test
+	public void getDiagnosisAttributeByUuid_shouldGetTheDiagnosisAttributeWithTheProvidedUuid() {
+		DiagnosisAttribute diagnosisAttribute = diagnosisService.getDiagnosisAttributeByUuid("31f7c3cd-699b-4ed3-af10-563e024cae76");
+		assertEquals("Testing Reference", diagnosisAttribute.getValueReference());
+		assertEquals(1, diagnosisAttribute.getId());
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getDiagnosisAttributeByUuid(String)
+	 */
+	@Test
+	public void getDiagnosisAttributeByUuid_shouldReturnNullIfNoDiagnosisAttributeHasTheProvidedUuid() {
+		assertNull(diagnosisService.getDiagnosisAttributeByUuid("001612a5-9f2a-4a05-9384-755679f75647"));
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#save(Diagnosis)
+	 */
+	@Test
+	public void saveDiagnosis_shouldSaveTheDiagnosisWithTheProvidedAttributes() {
+		final String NAMESPACE = "namespace", FORMFIELD_PATH = "formFieldPath";
+		final DiagnosisAttributeType diagnosisAttributeType = Context.getDiagnosisService()
+				.getDiagnosisAttributeTypeByUuid("949daf5b-a83e-4b65-b914-502a553243d3");
+		DiagnosisAttribute diagnosisAttribute = new DiagnosisAttribute();
+		diagnosisAttribute.setId(2);
+		diagnosisAttribute.setUuid("3cc9ad27-a4e1-4a08-acd5-fb6b847d71d0");
+		diagnosisAttribute.setAttributeType(diagnosisAttributeType);
+		diagnosisAttribute.setCreator(diagnosisAttributeType.getCreator());
+		diagnosisAttribute.setVoided(false);
+		diagnosisAttribute.setValueReferenceInternal("Diagnosis Attribute Reference");
+		String uuid = "3c6422d8-7bdd-4e20-bd1b-a590f084db08";
+		Condition condition = conditionService.getConditionByUuid("2cc6880e-2c46-15e4-9038-a6c5e4d22fb7");
+		Encounter encounter = encounterService.getEncounterByUuid("y403fafb-e5e4-42d0-9d11-4f52e89d123r");
+		Patient patient = patientService.getPatient(2);
+		Diagnosis diagnosis = new Diagnosis();
+		diagnosis.setUuid(uuid);
```

Comment:
Same for the uuid above.

Reply:
Or did you set the diagnosis uuid simply for the sake of being able to fetch it again using this uuid? If this is the case, then it is fine. :)

Reply:
Yes, I did not want to add an entry to the dataset.